src/codewhisperer/activation.ts

```diff
@@ -244,17 +244,15 @@ export async function activate(context: ExtContext): Promise<void> {
                 context.extensionContext.globalState.get<boolean>(
                     CodeWhispererConstants.accessTokenMigrationDoNotShowAgainKey
                 ) || false
-            const notificationLastShown =
-                context.extensionContext.globalState.get<Date>(
+            const notificationLastShown: number =
```

Comment:
I'd leave the type as `any`. Because existing state will be string, the logic after this needs to check `typeof(notificationLastShown)` and if it's a string, then default to `1` or something.

Reply:
We updated the key so it will save under that new key and will either be `undefined` or a number

Reply:
> globalState.get<number> makes no guarantees about the value at runtime.

Yup. The correct way to do this is actually `globalState.get<unknown>`. I usually don't mention `unknown` though because it requires even more explanation on how to convert 'unknown' types without using `as`

Reply:
The type is still incorrect though. If you remove the explicit type annotation, typescript inference will infer it as `number|undefined` since that is what you gave in the `get<>()` call.

It looks like `notificationLastShown +` addition operation is being done on a possibly-undefined value below.

```suggestion
            const notificationLastShown =
```

Reply:
oops I see the `|| 1` ensures `number`.

Reply:
It's because the code is still using `||` which is bad practice IMO. The correct way is:
```ts
               context.extensionContext.globalState.get<number>(
                    CodeWhispererConstants.accessTokenMigrationDoNotShowAgainLastShown,
                    1
                )
```
Explicit type annotations on variable declarations is not the same as a type assertion unlike `as` or type parameters so having `const notificationLastShown: number` is fine albeit verbose.